src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -1209,7 +1209,9 @@ void primitive_inst::do_runtime_in_place_kv_cache() {
     auto max_pad = kv_cache_inst::get_max_pad(past_layout, _deps[0].first->_max_output_layout_count[0], sequence_axis_legacy, "past_layout");
 
     if (max_pad > 0) {
-        kv_cache_inst::update_pad(present_layout, max_pad - 1, sequence_axis_legacy);
+        const auto past_seq_len = static_cast<int64_t>(past_layout.get_shape()[sequence_axis]);
+        const auto present_seq_len = static_cast<int64_t>(present_layout.get_shape()[sequence_axis]);
+        kv_cache_inst::update_pad(present_layout, max_pad - (present_seq_len - past_seq_len), sequence_axis_legacy);
```

Comment:
Umm now I think we need to do like this 
1. new_seq_len = input[1]'s seq_len (it is same as ( present_Seq_len - past_seq_len),  because present_seq_len = past_seq_len + input[1]'s seq_len)
2. if (max_pad - new_seq_len > 0) 
 update_pad(present_layout, max_pad - new_seq_len, sequence_axis_legacy)

Reply:
applied